src/Orleans/Core/IGrainInvokeInterceptor.cs

```diff
@@ -1,11 +1,18 @@
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Orleans.CodeGeneration;
+
 namespace Orleans
 {
-    using System.Reflection;
-    using System.Threading.Tasks;
-    using Orleans.CodeGeneration;
-
+    [Obsolete("Use IMethodInvocationInterceptor instead. This interface may be removed in a future release.")]
     public interface IGrainInvokeInterceptor
     {
         Task<object> Invoke(MethodInfo method, InvokeMethodRequest request, IGrainMethodInvoker invoker);
     }
+
+    public interface IGrainCallFilter
```

Comment:
either move to a different file (preferable) or at least rename the file, since this is now the important interface for interception